pkgs/applications/audio/renoise/default.nix

```diff
@@ -0,0 +1,83 @@
+{ stdenv, lib, requireFile, demo, fetchurl, libX11, libXext, libXcursor, libXrandr, alsaLib, ... }:
+
+let
+  unlines = strs: lib.concatStrings ((map (x: x + "\n")) strs);
+
+  renoise =
+    stdenv.mkDerivation {
+      name = "renoise";
+
+      phases = "unpackPhase installPhase";
+
+      src =
+        if builtins.currentSystem == "x86_64-linux" then
+          if demo then
+            fetchurl {
+              url = "http://files.renoise.com/demo/Renoise_3_0_1_Demo_x86_64.tar.bz2";
+              sha256 = "1q7f94wz2dbz659kpp53a3n1qyndsk0pkb29lxdff4pc3ddqwykg";
+            }
+          else
+            requireFile {
+              url = "http://backstage.renoise.com/frontend/app/index.html#/login";
+              name = "rns_3_0_1_linux_x86_64.tar.gz";
+              sha256 = "0n23m0f3pkwgicgvvg4hr7wa1jdk0py91fndda5986ri0km4j6af";
+            }
+        else throw "haven't written this for systems besides 64-bit linux";
```

Comment:
I _guess_ you can remove the `if current system is x86_64` thing by providing `meta.platforms`, but I'm not sure whether this keeps a non-x86_64 user from installing this!